common/metrics/structured/doc.go

```diff
@@ -0,0 +1,119 @@
+to the new data you wanted to add.
+
+# To see what tags an existing metric has
+
+Find the name string (e.g. grep for it), open it in an IDE, and just ask the
+IDE to auto-complete a field access:
+
+	yourTagsInstance.<ctrl-space to request autocomplete>
+
+In Goland, VSCode, and likely elsewhere, this will give you a drop-down of all
+fields inherited from all parents, for easy reading.
+
+# Best practices
+
+Use constant, in-line strings for metric names.  Prometheus requires that each
+"name" must have a stable set of tags, so there is no safety benefit to using a
+const - generally speaking it must NOT be shared.
+
+Ad-hoc metrics are encouraged to use the convenience methods for simplicity.
+When curious about something, just emit a metric and find out later (but watch
+out for cardinality).
+
+Avoid pointers, both for the ...Tags struct and its values, to prevent mutation.
+This also implies you should generally use "simple" and minimal field types, as
+they will be copied repeatedly - avoid e.g. complex thrift objects.  Hopefully
+this will end up being nicer to the garbage collector than pointers everywhere.
+
+For any metrics (or "events" which have multiple metrics) you consider "stable"
+or have alerts or dashboards based on, strongly consider declaring a method on
+your ...Tags struct and emitting in there.  This helps inform reviewers that
+changing the metrics might cause problems elsewhere, and documents intent for
+Cadence operators if they get an alert or see strange numbers.
+
+# Code generation customization
+
+Fields have two major options available: they can declare a custom to-string
+conversion, and they can "reserve" a tag without defining a value:
+
+	type YourTags struct {
+		Fancy protobuf.Thing `tag:"fancy" convert:"{{.}}.String()"`
+		Reserved struct{} `tag:"reserved"`
+	}
+
+Custom conversion is just a text/template string, where `.` will be filled in
+with the field access (i.e. `y.Fancy`).  Strings work automatically, and
+integers (int, int32, and int64) will be automatically `strconv.Itoa`-converted,
+but all other types will require custom conversion.  As you cannot declare new
+imports in this string, make sure you've imported any packages you need to
+stringify a value in the same file as the ...Tags is declared.
+
+Reserved tags serve two purposes:
+  - They document that a tag will be emitted, so it can be discovered
+  - They reserve space in the map returned by `GetTags()`, so you can
+    efficiently add it at runtime
+
+Because reserved tags will not be filled in by convenience methods like `Count`,
+they are almost exclusively useful for methods that emit specific metrics.
+
+For the simplest use cases, use a method on the ...Tags struct and add the tags
+by hand:
+
+	func (s SomethingTags) ItHappened(times int) {
+		tags := s.GetTags()                          // get all static tags
+		tags["reserved"] = fmt.Sprint(rand.Intn(10)) // add the reserved one(s)
```

Comment:
I assume you mean to give an exmaple of falling into 1 of 10 buckets, it's a little confusing at first, I'd just used a fixed value to reduce confusion... 

or alternatively I admit I don't follow the intent of the tag here